internal/cluster/common.go

```diff
@@ -80,6 +80,33 @@ func updateClusterStatus(ctx context.Context, log logrus.FieldLogger, db *gorm.D
 	return cluster, nil
 }
 
+func completeKubeconfigAvailableProgress(log logrus.FieldLogger, db *gorm.DB, c *common.Cluster) (*common.Cluster, error) {
```

Comment:
Maybe if we move this call to somewhere else we don't have to do the entire DB transaction from scratch?

Reply:
I am open to suggestions. Any particular location in mind?
I didn't want to include it in `clusterAPI.CompleteInstallation`, which gets called before this function, for two reasons:
1. It adds a non-obvious (IMO) scope to that API, and it did not seem to belong there.
2. `clusterAPI.CompleteInstallation` uses a generic function named `updateClusterStatus` to update the status, I wanted to avoid from changing it.

Reply:
Looking at updateClusterStatus again, I see it already takes a decision internally to`update install_completed_at`based on the dest status:
https://github.com/openshift/assisted-service/blob/e54185050dd372cabd3167fd0667818fe1770cef/internal/cluster/common.go#L59

No reason why we can’t do the same with progress. Updated in this revision.

Reply:
this will also make the status and progress update in a single DB transaction, as requested in another comment.